apps/meeting/src/containers/MeetingProviderWrapper/index.tsx

```diff
@@ -17,6 +23,29 @@ import MeetingEventObserver from '../MeetingEventObserver';
 import meetingConfig from '../../meetingConfig';
 import { useAppState } from '../../providers/AppStateProvider';
 
+const MeetingProviderWithDeviceReplacement: React.FC = ({ children }) => {
+  const { isWebAudioEnabled } = useAppState();
+  const { addVoiceFocus } = useVoiceFocus();
+
+  const replaceDevice = (
+    nextDevice: string, 
+    currentDevice: Device | AudioTransformDevice
+  ): Promise<Device | VoiceFocusTransformDevice> => {
+    if (currentDevice instanceof VoiceFocusTransformDevice) {
+      return addVoiceFocus(nextDevice);
+    }
+    return Promise.resolve(nextDevice);
+  };
+
+  const meetingConfigValue = {
+    ...meetingConfig,
+    enableWebAudio: isWebAudioEnabled,
+    onDeviceReplacement: isWebAudioEnabled ? replaceDevice : undefined
```

Comment:
If we never support turning off web audio — and I suggest we should not, because it won't work — I think you can simplify this. You check for web audio when deciding which provider to build, so you don't have to inside the provider.

Reply:
Good catch! Will set `true` to `enableWebAudio`, and `replaceDevice` to `onDeviceReplacement`